forc-plugins/forc-client/src/op/run/mod.rs

```diff
@@ -177,24 +175,34 @@ async fn send_tx(
 ) -> Result<Vec<fuel_tx::Receipt>> {
     let outputs = {
         if !simulate {
-            let (_, receipts) = client.submit_and_await_commit_with_receipts(tx).await?;
-            if let Some(receipts) = receipts {
-                Ok(receipts)
-            } else {
-                bail!("The `receipts` during `send_tx` is empty")
+            let status = client.submit_and_await_commit(tx).await?;
+
+            match status {
+                fuel_core_client::client::types::TransactionStatus::Success {
+                    receipts, ..
+                } => receipts,
+                fuel_core_client::client::types::TransactionStatus::Failure {
+                    receipts, ..
+                } => receipts,
+                _ => vec![],
             }
         } else {
-            client.dry_run(tx).await
+            let txs = vec![tx.clone()];
+            let receipts = client.dry_run(txs.as_slice()).await?;
+            let receipts = receipts
+                .first()
+                .map(|tx| &tx.result)
+                .map(|res| res.receipts());
+            match receipts {
+                Some(receipts) => receipts.to_vec(),
+                None => vec![],
+            }
         }
     };
-
-    match outputs {
-        Ok(logs) => {
-            info!("{}", format_log_receipts(&logs, pretty_print)?);
-            Ok(logs)
-        }
-        Err(e) => bail!("{e}"),
+    if !outputs.is_empty() {
+        info!("{}", format_log_receipts(&outputs, pretty_print)?);
     }
```

Comment:
Should we still bail when receipts are empty?

Reply:
I think output was returning a Result but not anymore, so even if we had empty receipts we were not bailing. So it should be the same behaviour now but I might be missing something.